docs/contributing/backward-compatibility-promise.md

```diff
@@ -0,0 +1,75 @@
+# Backward Compatibility Promise
+
+Smooth and safe upgrades of your own e-commerce project are very important to us.
+In the same time, we need to be able to improve Shopsys Framework for you by adding functionality, enhancing or simplifying current functions and fixing bugs.
+After reading this promise you'll understand backward compatibility, what changes you can expect and how we plan to make changes in the future.
+
+*Note: This BC promise becomes effective with the release first non-beta release (`7.0.0`).*
+
+## Releases and Versioning
+This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html), which means it's release versions are in format `MAJOR.MINOR.PATCH`:
+
+- `MAJOR` version may contain incompatible changes
+- `MINOR` version may add new functionality in a backward-compatible manner
+- `PATCH` version contains only backward-compatible bug fixes
```

Comment:
Imagine a situation:
* we release a version (major/minor)
* after couple of days we find a bug
* the solution for the bug is backward-**incompatible**

What to do in such situation? I think this situation can happen and we should be prepared for it

Reply:
Well, if it needs to be fixed and cannot be done backward-compatible we'd still have to do it. Maybe let's add a mention about necessary BC breaks for security or critical bugfixes, which would have to include `[BC-BREAK]` in the PR description (meaning it would be included in `CHAGNELOG.md`) and the reason for making the BC break would have to be explained in the description.

Reply:
Or you can write a *Note* or *Blockquote* that would explain something like "We are doing the best job we can, but we want to be prepared also for situation when it will be necessary to release a patch version that breaks backward compatibility. If this situation occurs, we'll inform you together with instruction what to do"

And you don't have to create rules what will be in PR and so on.

Reply:
I've added the note in 58d3a8fcd70adb4cc275b69366c47a77cd533463